crates/crates_io_trustpub/src/github/workflows.rs

```diff
@@ -0,0 +1,109 @@
+use std::sync::LazyLock;
+
+/// Extracts the workflow filename from a GitHub workflow reference.
+///
+/// In other words, it turns e.g. `rust-lang/regex/.github/workflows/ci.yml@refs/heads/main`
+/// into `ci.yml`, or `None` if the reference is in an unexpected format.
+pub(crate) fn extract_workflow_filename(workflow_ref: &str) -> Option<&str> {
+    static WORKFLOW_REF_RE: LazyLock<regex::Regex> =
+        LazyLock::new(|| regex::Regex::new(r"([^/]+\.(yml|yaml))(@.+)").unwrap());
```

Comment:
PyPI uses a "lookahead match" pattern, but that does not appear to be supported by the `regex` crate. I've slightly adjusted the pattern to make it work, but since all the test cases from PyPI are passing I'm fairly confident in the change.